Bottom: b20a5da6c8b8f2d326e20f9fc7922050d404fe2b
Top:    79ea8d8e37e054372ed674482eb484c01524e58a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 20:35:12 +0000

Draw lines as their polygon equivalent (helps check for bugs in their outline representation.


---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 95d06b3..5ade074 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -1,3 +1,5 @@
+#include <string.h>
+
 #include "global.h"
 #include "hid.h"
 #include "hid_draw.h"
@@ -10,6 +12,13 @@
 static void
 common_draw_pcb_line (hidGC gc, LineType *line)
 {
+  PolygonType poly;
+  memset (&poly, 0, sizeof (PolygonType));
+  poly.Clipped = LinePoly (line, line->Thickness);
+  hid_draw_pcb_polygon (gc, &poly, NULL);
+  poly_Free (&poly.Clipped);
+  return;
+
   hid_draw_set_line_cap (gc, Trace_Cap);
   if (TEST_FLAG (THINDRAWFLAG, PCB))
     hid_draw_set_line_width (gc, 0);
